### PR TITLE
stackoverflow_survey_2024

### PR DESCRIPTION
**Pull request by havishak**
Thank you for your submission! Please leave this text as-is when you submit, and then check the checklist.

Usability:

- [X] This dataset has not already been used in TidyTuesday.
- [X] The dataset is less than 20MB when saved as a tidy CSV.
- [X] I can imagine at least one data visualization related to this dataset.

Preparation:

- [] `cleaning.R` has code to download and clean the dataset, resulting in descriptively-named data.frames.
- [X] I ran `saving.R` to create `csv`s and `md` data dictionaries for each of my data.frames.
- [X] I filled in descriptions for each variable in each of the `{dataset}.md` files.
- [X] I edited the `intro.md` file to introduce my dataset.
- [X] I included at least one image for my dataset as a `png` file.
- [X] I completed the information in `meta.yaml`, including descriptive alt text for each image.
- [X] I provided information in `meta.yaml` about how to credit me, and deleted any parts of that block that I do not want you to use.

### CODE REVIEW

**Comment by jonthegeek** [comment]
Thank you so much for this submission! I'm traveling and won't have a chance to review it right away, but I'll get to it as soon as I can. Thanks again!

File: data/curated/stackoverflow_survey_2024/stackoverflow_survey_single_response.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+| main_branch           | integer   | Which of the following options best describes you today?                            |
+| age                   | integer   | Age                                                                                 |
+| remote_work           | integer   | Current work situation                                                              |
+| ed_level              | integer   | Highest education level completed                                                   |
+| years_code            | integer   | Years have you been coding in total; More than 50 years coded as 51                 |
+| years_code_pro        | integer   | Years have you coded professionally; More than 50 years coded as 51                 |
+| dev_type              | integer   | Best current-job description                                                        |
+| org_size              | integer   | People in the organization                                                          |
+| purchase_influence    | integer   | Level of influence in purchasing new technology at your organization                |
```
Comment (jonthegeek) [inline data/curated/stackoverflow_survey_2024/stackoverflow_survey_single_response.md:12]:
```suggestion
| purchase_influence    | integer   | Level of influence in purchasing new technology at their organization                |
```

File: data/curated/stackoverflow_survey_2024/stackoverflow_survey_single_response.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+| variable              | class     | description                                                                         |
+|:--------------------------|:-----------------|:--------------------------|
+| response_id           | double    | Respondent ID                                                                       |
+| main_branch           | integer   | Which of the following options best describes you today?                            |
```
Comment (jonthegeek) [inline data/curated/stackoverflow_survey_2024/stackoverflow_survey_single_response.md:4]:
```suggestion
| main_branch           | integer   | Professional coding level of the respondent |
```

File: data/curated/stackoverflow_survey_2024/stackoverflow_survey_single_response.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+| remote_work           | integer   | Current work situation                                                              |
+| ed_level              | integer   | Highest education level completed                                                   |
+| years_code            | integer   | Years have you been coding in total; More than 50 years coded as 51                 |
+| years_code_pro        | integer   | Years have you coded professionally; More than 50 years coded as 51                 |
+| dev_type              | integer   | Best current-job description                                                        |
+| org_size              | integer   | People in the organization                                                          |
+| purchase_influence    | integer   | Level of influence in purchasing new technology at your organization                |
+| buildvs_buy           | integer   | When asked to provide a recommendation, what has been your most recent choice?      |
+| country               | character | Country Living In                                                                   |
```
Comment (jonthegeek) [inline data/curated/stackoverflow_survey_2024/stackoverflow_survey_single_response.md:14]:
```suggestion
| country               | character | Country in which the respondent lives |
```

File: data/curated/stackoverflow_survey_2024/stackoverflow_survey_single_response.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+| age                   | integer   | Age                                                                                 |
+| remote_work           | integer   | Current work situation                                                              |
+| ed_level              | integer   | Highest education level completed                                                   |
+| years_code            | integer   | Years have you been coding in total; More than 50 years coded as 51                 |
+| years_code_pro        | integer   | Years have you coded professionally; More than 50 years coded as 51                 |
+| dev_type              | integer   | Best current-job description                                                        |
+| org_size              | integer   | People in the organization                                                          |
+| purchase_influence    | integer   | Level of influence in purchasing new technology at your organization                |
+| buildvs_buy           | integer   | When asked to provide a recommendation, what has been your most recent choice?      |
```
Comment (jonthegeek) [inline data/curated/stackoverflow_survey_2024/stackoverflow_survey_single_response.md:13]:
```suggestion
| buildvs_buy           | integer   | How much customization was needed in most recent tool recommendation |
```

File: data/curated/stackoverflow_survey_2024/stackoverflow_survey_single_response.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+| country               | character | Country Living In                                                                   |
+| currency              | character | Currency of the country                                                             |
+| comp_total            | double    | Total compensation                                                                  |
+| so_visit_freq         | integer   | Stack Overflow visiting frequency                                                   |
+| so_account            | integer   | Stack Overflow Account Status                                                       |
```
Comment (jonthegeek) [inline data/curated/stackoverflow_survey_2024/stackoverflow_survey_single_response.md:18]:
```suggestion
| so_account            | integer   | Stack Overflow account status                                                       |
```

File: data/curated/stackoverflow_survey_2024/stackoverflow_survey_single_response.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+| so_part_freq          | integer   | Stack Overflow participation frequency                                              |
+| so_comm               | integer   | Do you consider yourself a member of the Stack Overflow community?                  |
```
Comment (jonthegeek) [inline data/curated/stackoverflow_survey_2024/stackoverflow_survey_single_response.md:20]:
```suggestion
| so_comm               | integer   | Whether the respondent considers themself a member of the Stack Overflow community?                  |
```

File: data/curated/stackoverflow_survey_2024/stackoverflow_survey_single_response.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+| ai_select             | integer   | Use AI in development process                                                       |
+| ai_sent               | integer   | Stance on using AI tools as part of development workflow                            |
+| ai_acc                | integer   | Trust in accuracy of AI as part of development workflow                             |
+| ai_complex            | integer   | How well do the AI tools you use in your development workflow handle complex tasks? |
```
Comment (jonthegeek) [inline data/curated/stackoverflow_survey_2024/stackoverflow_survey_single_response.md:24]:
```suggestion
| ai_complex            | integer   | How well the respondent believes the AI tools they use in development workflows handle complex tasks |
```

File: data/curated/stackoverflow_survey_2024/stackoverflow_survey_single_response.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+| ai_threat             | integer   | Do you believe AI is a threat to your current job?                                  |
+| survey_length         | integer   | How do you feel about the length of the survey this year?                           |
+| survey_ease           | integer   | How easy or difficult was this survey to complete?                                  |
```
Comment (jonthegeek) [inline data/curated/stackoverflow_survey_2024/stackoverflow_survey_single_response.md:27]:
```suggestion
| ai_threat             | integer   | Belief that AI is a threat to current job |
| survey_length         | integer   | Feeling about the length of the Stack Overflow Developer Survey this year |
| survey_ease           | integer   | Ease of completion of this survey |
```

**Review by jonthegeek — APPROVED** [review]
Thank you so much! I made some tweaks (which I'll apply in a moment) and added a dataset with the questions, but this was extremely helpful... particularly since I'm about to leave on another trip and wouldn't have time to prepare a dataset! Thanks again!

File: data/2024/2024-09-03/readme.md [hunk 9]
```diff
@@ -0,0 +1,259 @@
...
+4.  Stack Overflow community
+5.  Artificial Intelligence  (AI)
+6.  Professional Developer Series - Not part of the main survey
+7.  Thoughts on Survey
+
+The dataset provided for this analysis focuses exclusively on the single-response questions from the main survey sections. Each categorical response in the survey has been integer-coded, with corresponding labels available in the crosswalk file.
+
+What can you see about developer demographics? How do developers engage with Stack Overflow? What do they think about AI?
+
+Thank you to [credit](https://github.com/havishak) for curating this week's dataset.
```
Comment (jonthegeek) [inline data/2024/2024-09-03/readme.md:17]:
```suggestion
Thank you to [Havisha Khurana](https://github.com/havishak) for curating this week's dataset!
```

File: static/tt_data_type.csv [hunk 10]
```diff
@@ -1,4 +1,7 @@
 Week,Date,year,data_files,data_type,delim
+36,2024-09-03,2024,qname_levels_single_response_crosswalk.csv,csv,","
+36,2024-09-03,2024,stackoverflow_survey_questions.csv,csv,","
+36,2024-09-03,2024,stackoverflow_survey_single_response.csv,csv,","
 34,2024-08-20,2024,english_monarchs_marriages_df.csv,csv,","
```
Comment (jonthegeek) [inline static/tt_data_type.csv:5]:
```suggestion
35,2024-08-27,2024,power_rangers_episodes.csv,csv,","
35,2024-08-27,2024,power_rangers_seasons.csv,csv,","
34,2024-08-20,2024,english_monarchs_marriages_df.csv,csv,","
```

**Review by jonthegeek — APPROVED** [review]
We found a couple bugs in my system! Thanks for helping to beta test this process!

**Comment by lgibson7** [comment]
Thanks @havishak!

**Comment by havishak** [comment]
Thanks for the tweaks @jonthegeek! I felt excited about contributing!!

**Pull request merged**